File: src/main/java/ch/uzh/ifi/access/model/dto/TaskDTO.java
package ch.uzh.ifi.access.model.dto;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.HashMap;
import java.util.Map;

@Data
@NoArgsConstructor
public class TaskDTO {

    public String slug;

    public Integer ordinalNum;

    public Map<String,TaskInformationDTO> information = new HashMap<>();

    public Double maxPoints;

    public Integer maxAttempts;

    public Integer refill;

    public TaskEvaluatorDTO evaluator;

    public TaskFilesDTO files;
}

File: src/main/kotlin/ch/uzh/ifi/access/model/dto/TaskDTO.kt
package ch.uzh.ifi.access.model.dto

import lombok.Data

@Data
class TaskDTO(
    var slug: String? = null,
    var ordinalNum: Int? = null,
    var information: MutableMap<String, ProblemInformationDTO> = HashMap(),
    var maxPoints: Double? = null,
    var maxAttempts: Int? = null,
    var refill: Int? = null,
    var evaluator: ProblemEvaluatorDTO? = null,
    var files: TaskFilesDTO? = null
)
